ql/src/java/org/apache/hadoop/hive/ql/parse/QB.java

```diff
@@ -506,4 +511,15 @@ public void addSubqExprAlias(ASTNode expressionTree, SemanticAnalyzer semanticAn
     aliasToSubqExpr.put(alias, qbexpr);
     addAlias(alias);
   }
+
+  public void iterate(QBVisitor qbVisitor) {
```

Comment:
not sure we need this in order to extract table names, see `ctx.getParsedTables()`
cc @kasakrisz

Reply:
getParsedTables seems to works as told in another comment, I'm considering this resolved